CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+# How to contribute
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Fork the repository on GitHub
+
+## Making Changes
+
+* Create a topic branch from where you want to base your work (this is almost
+definitely the master branch).
+  * To quickly create a topic branch based on master; `git branch
+    fix/master/my_contribution master` then checkout the new branch with `git
+    checkout fix/master/my_contribution`.
+  * Please avoid working directly on the
+    `master` branch.
+* Make commits of logical units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the proper format.
+
+````
+    Make the example in CONTRIBUTING imperative and concrete
```

Comment:
I might instead remove the bit that mentions a ticket number from the commit message? Since the rest of this doesn't mention requiring creation of a JIRA ticket, and I don't think we're at a point where we're getting so many contributions right now that it makes sense to require that.